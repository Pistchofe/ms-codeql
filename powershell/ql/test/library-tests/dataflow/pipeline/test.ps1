function produce {
    $x = Source "16"
    $y = Source "17"
    $z = Source "18"
    $x
    $y, $z
}

function consume {
    Param([Parameter(ValueFromPipeline)] $x)

    process {
        Sink $x # $ hasValueFlow=16 hasValueFlow=17 hasValueFlow=18 hasValueFlow=19 hasValueFlow=20
    }
}

produce | consume

$x = Source "19"
$y = Source "20"
$x, $y | consume

function consume2 {
    process {
        Sink $_ # $ hasValueFlow=21 hasValueFlow=22
    }
}

$x = Source "21"
$y = Source "22"
$x, $y | consume2