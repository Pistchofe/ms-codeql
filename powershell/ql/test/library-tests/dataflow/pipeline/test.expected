models
edges
| test.ps1:2:10:2:20 | Source | test.ps1:5:5:5:7 | x | provenance |  |
| test.ps1:3:10:3:20 | Source | test.ps1:6:5:6:7 | y | provenance |  |
| test.ps1:4:10:4:20 | Source | test.ps1:6:9:6:11 | z | provenance |  |
| test.ps1:5:5:5:7 | x | test.ps1:17:1:17:8 | produce [element] | provenance |  |
| test.ps1:6:5:6:7 | y | test.ps1:6:5:6:11 | ...,... [element 0] | provenance |  |
| test.ps1:6:5:6:11 | ...,... [element 0] | test.ps1:17:1:17:8 | produce [element] | provenance |  |
| test.ps1:6:5:6:11 | ...,... [element 1] | test.ps1:17:1:17:8 | produce [element] | provenance |  |
| test.ps1:6:9:6:11 | z | test.ps1:6:5:6:11 | ...,... [element 1] | provenance |  |
| test.ps1:10:11:10:44 | x [element 0] | test.ps1:13:14:13:16 | x | provenance |  |
| test.ps1:10:11:10:44 | x [element 1] | test.ps1:13:14:13:16 | x | provenance |  |
| test.ps1:10:11:10:44 | x [element] | test.ps1:13:14:13:16 | x | provenance |  |
| test.ps1:17:1:17:8 | produce [element] | test.ps1:10:11:10:44 | x [element] | provenance |  |
| test.ps1:19:6:19:16 | Source | test.ps1:21:1:21:3 | x | provenance |  |
| test.ps1:20:6:20:16 | Source | test.ps1:21:5:21:7 | y | provenance |  |
| test.ps1:21:1:21:3 | x | test.ps1:21:1:21:7 | ...,... [element 0] | provenance |  |
| test.ps1:21:1:21:7 | ...,... [element 0] | test.ps1:10:11:10:44 | x [element 0] | provenance |  |
| test.ps1:21:1:21:7 | ...,... [element 0] | test.ps1:21:1:21:7 | ...,... [element 0] | provenance |  |
| test.ps1:21:1:21:7 | ...,... [element 1] | test.ps1:10:11:10:44 | x [element 1] | provenance |  |
| test.ps1:21:1:21:7 | ...,... [element 1] | test.ps1:21:1:21:7 | ...,... [element 1] | provenance |  |
| test.ps1:21:5:21:7 | y | test.ps1:21:1:21:7 | ...,... [element 1] | provenance |  |
| test.ps1:25:14:25:16 | _ [element 0] | test.ps1:25:14:25:16 | _ | provenance |  |
| test.ps1:25:14:25:16 | _ [element 1] | test.ps1:25:14:25:16 | _ | provenance |  |
| test.ps1:29:6:29:16 | Source | test.ps1:31:1:31:3 | x | provenance |  |
| test.ps1:30:6:30:16 | Source | test.ps1:31:5:31:7 | y | provenance |  |
| test.ps1:31:1:31:3 | x | test.ps1:31:1:31:7 | ...,... [element 0] | provenance |  |
| test.ps1:31:1:31:7 | ...,... [element 0] | test.ps1:25:14:25:16 | _ [element 0] | provenance |  |
| test.ps1:31:1:31:7 | ...,... [element 0] | test.ps1:31:1:31:7 | ...,... [element 0] | provenance |  |
| test.ps1:31:1:31:7 | ...,... [element 1] | test.ps1:25:14:25:16 | _ [element 1] | provenance |  |
| test.ps1:31:1:31:7 | ...,... [element 1] | test.ps1:31:1:31:7 | ...,... [element 1] | provenance |  |
| test.ps1:31:5:31:7 | y | test.ps1:31:1:31:7 | ...,... [element 1] | provenance |  |
| test.ps1:34:11:34:58 | x [element x] | test.ps1:36:10:36:12 | x | provenance |  |
| test.ps1:39:6:39:16 | Source | test.ps1:40:23:40:25 | x | provenance |  |
| test.ps1:40:1:40:26 | [...]... [element x] | test.ps1:34:11:34:58 | x [element x] | provenance |  |
| test.ps1:40:17:40:26 | ${...} [element x] | test.ps1:40:1:40:26 | [...]... [element x] | provenance |  |
| test.ps1:40:23:40:25 | x | test.ps1:40:17:40:26 | ${...} [element x] | provenance |  |
| test.ps1:43:11:43:58 | x [element 0, element x] | test.ps1:46:14:46:16 | x | provenance |  |
| test.ps1:43:11:43:58 | x [element 1, element x] | test.ps1:46:14:46:16 | x | provenance |  |
| test.ps1:43:11:43:58 | x [element 2, element x] | test.ps1:46:14:46:16 | x | provenance |  |
| test.ps1:50:1:50:34 | [...]... [element x] | test.ps1:50:1:50:106 | ...,... [element 0, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 0, element x] | test.ps1:43:11:43:58 | x [element 0, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 0, element x] | test.ps1:50:1:50:106 | ...,... [element 0, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 1, element x] | test.ps1:43:11:43:58 | x [element 1, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 1, element x] | test.ps1:50:1:50:106 | ...,... [element 1, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 2, element x] | test.ps1:43:11:43:58 | x [element 2, element x] | provenance |  |
| test.ps1:50:1:50:106 | ...,... [element 2, element x] | test.ps1:50:1:50:106 | ...,... [element 2, element x] | provenance |  |
| test.ps1:50:17:50:34 | ${...} [element x] | test.ps1:50:1:50:34 | [...]... [element x] | provenance |  |
| test.ps1:50:23:50:33 | Source | test.ps1:50:17:50:34 | ${...} [element x] | provenance |  |
| test.ps1:50:36:50:70 | [...]... [element x] | test.ps1:50:1:50:106 | ...,... [element 1, element x] | provenance |  |
| test.ps1:50:52:50:70 | ${...} [element x] | test.ps1:50:36:50:70 | [...]... [element x] | provenance |  |
| test.ps1:50:58:50:69 | Source | test.ps1:50:52:50:70 | ${...} [element x] | provenance |  |
| test.ps1:50:72:50:106 | [...]... [element x] | test.ps1:50:1:50:106 | ...,... [element 2, element x] | provenance |  |
| test.ps1:50:88:50:106 | ${...} [element x] | test.ps1:50:72:50:106 | [...]... [element x] | provenance |  |
| test.ps1:50:94:50:105 | Source | test.ps1:50:88:50:106 | ${...} [element x] | provenance |  |
nodes
| test.ps1:2:10:2:20 | Source | semmle.label | Source |
| test.ps1:3:10:3:20 | Source | semmle.label | Source |
| test.ps1:4:10:4:20 | Source | semmle.label | Source |
| test.ps1:5:5:5:7 | x | semmle.label | x |
| test.ps1:6:5:6:7 | y | semmle.label | y |
| test.ps1:6:5:6:11 | ...,... [element 0] | semmle.label | ...,... [element 0] |
| test.ps1:6:5:6:11 | ...,... [element 1] | semmle.label | ...,... [element 1] |
| test.ps1:6:9:6:11 | z | semmle.label | z |
| test.ps1:10:11:10:44 | x [element 0] | semmle.label | x [element 0] |
| test.ps1:10:11:10:44 | x [element 1] | semmle.label | x [element 1] |
| test.ps1:10:11:10:44 | x [element] | semmle.label | x [element] |
| test.ps1:13:14:13:16 | x | semmle.label | x |
| test.ps1:17:1:17:8 | produce [element] | semmle.label | produce [element] |
| test.ps1:19:6:19:16 | Source | semmle.label | Source |
| test.ps1:20:6:20:16 | Source | semmle.label | Source |
| test.ps1:21:1:21:3 | x | semmle.label | x |
| test.ps1:21:1:21:7 | ...,... [element 0] | semmle.label | ...,... [element 0] |
| test.ps1:21:1:21:7 | ...,... [element 0] | semmle.label | ...,... [element 0] |
| test.ps1:21:1:21:7 | ...,... [element 1] | semmle.label | ...,... [element 1] |
| test.ps1:21:1:21:7 | ...,... [element 1] | semmle.label | ...,... [element 1] |
| test.ps1:21:5:21:7 | y | semmle.label | y |
| test.ps1:25:14:25:16 | _ | semmle.label | _ |
| test.ps1:25:14:25:16 | _ [element 0] | semmle.label | _ [element 0] |
| test.ps1:25:14:25:16 | _ [element 1] | semmle.label | _ [element 1] |
| test.ps1:29:6:29:16 | Source | semmle.label | Source |
| test.ps1:30:6:30:16 | Source | semmle.label | Source |
| test.ps1:31:1:31:3 | x | semmle.label | x |
| test.ps1:31:1:31:7 | ...,... [element 0] | semmle.label | ...,... [element 0] |
| test.ps1:31:1:31:7 | ...,... [element 0] | semmle.label | ...,... [element 0] |
| test.ps1:31:1:31:7 | ...,... [element 1] | semmle.label | ...,... [element 1] |
| test.ps1:31:1:31:7 | ...,... [element 1] | semmle.label | ...,... [element 1] |
| test.ps1:31:5:31:7 | y | semmle.label | y |
| test.ps1:34:11:34:58 | x [element x] | semmle.label | x [element x] |
| test.ps1:36:10:36:12 | x | semmle.label | x |
| test.ps1:39:6:39:16 | Source | semmle.label | Source |
| test.ps1:40:1:40:26 | [...]... [element x] | semmle.label | [...]... [element x] |
| test.ps1:40:17:40:26 | ${...} [element x] | semmle.label | ${...} [element x] |
| test.ps1:40:23:40:25 | x | semmle.label | x |
| test.ps1:43:11:43:58 | x [element 0, element x] | semmle.label | x [element 0, element x] |
| test.ps1:43:11:43:58 | x [element 1, element x] | semmle.label | x [element 1, element x] |
| test.ps1:43:11:43:58 | x [element 2, element x] | semmle.label | x [element 2, element x] |
| test.ps1:46:14:46:16 | x | semmle.label | x |
| test.ps1:50:1:50:34 | [...]... [element x] | semmle.label | [...]... [element x] |
| test.ps1:50:1:50:106 | ...,... [element 0, element x] | semmle.label | ...,... [element 0, element x] |
| test.ps1:50:1:50:106 | ...,... [element 0, element x] | semmle.label | ...,... [element 0, element x] |
| test.ps1:50:1:50:106 | ...,... [element 1, element x] | semmle.label | ...,... [element 1, element x] |
| test.ps1:50:1:50:106 | ...,... [element 1, element x] | semmle.label | ...,... [element 1, element x] |
| test.ps1:50:1:50:106 | ...,... [element 2, element x] | semmle.label | ...,... [element 2, element x] |
| test.ps1:50:1:50:106 | ...,... [element 2, element x] | semmle.label | ...,... [element 2, element x] |
| test.ps1:50:17:50:34 | ${...} [element x] | semmle.label | ${...} [element x] |
| test.ps1:50:23:50:33 | Source | semmle.label | Source |
| test.ps1:50:36:50:70 | [...]... [element x] | semmle.label | [...]... [element x] |
| test.ps1:50:52:50:70 | ${...} [element x] | semmle.label | ${...} [element x] |
| test.ps1:50:58:50:69 | Source | semmle.label | Source |
| test.ps1:50:72:50:106 | [...]... [element x] | semmle.label | [...]... [element x] |
| test.ps1:50:88:50:106 | ${...} [element x] | semmle.label | ${...} [element x] |
| test.ps1:50:94:50:105 | Source | semmle.label | Source |
subpaths
testFailures
#select
| test.ps1:13:14:13:16 | x | test.ps1:2:10:2:20 | Source | test.ps1:13:14:13:16 | x | $@ | test.ps1:2:10:2:20 | Source | Source |
| test.ps1:13:14:13:16 | x | test.ps1:3:10:3:20 | Source | test.ps1:13:14:13:16 | x | $@ | test.ps1:3:10:3:20 | Source | Source |
| test.ps1:13:14:13:16 | x | test.ps1:4:10:4:20 | Source | test.ps1:13:14:13:16 | x | $@ | test.ps1:4:10:4:20 | Source | Source |
| test.ps1:13:14:13:16 | x | test.ps1:19:6:19:16 | Source | test.ps1:13:14:13:16 | x | $@ | test.ps1:19:6:19:16 | Source | Source |
| test.ps1:13:14:13:16 | x | test.ps1:20:6:20:16 | Source | test.ps1:13:14:13:16 | x | $@ | test.ps1:20:6:20:16 | Source | Source |
| test.ps1:25:14:25:16 | _ | test.ps1:29:6:29:16 | Source | test.ps1:25:14:25:16 | _ | $@ | test.ps1:29:6:29:16 | Source | Source |
| test.ps1:25:14:25:16 | _ | test.ps1:30:6:30:16 | Source | test.ps1:25:14:25:16 | _ | $@ | test.ps1:30:6:30:16 | Source | Source |
| test.ps1:36:10:36:12 | x | test.ps1:39:6:39:16 | Source | test.ps1:36:10:36:12 | x | $@ | test.ps1:39:6:39:16 | Source | Source |
| test.ps1:46:14:46:16 | x | test.ps1:50:23:50:33 | Source | test.ps1:46:14:46:16 | x | $@ | test.ps1:50:23:50:33 | Source | Source |
| test.ps1:46:14:46:16 | x | test.ps1:50:58:50:69 | Source | test.ps1:46:14:46:16 | x | $@ | test.ps1:50:58:50:69 | Source | Source |
| test.ps1:46:14:46:16 | x | test.ps1:50:94:50:105 | Source | test.ps1:46:14:46:16 | x | $@ | test.ps1:50:94:50:105 | Source | Source |
